main.go

```diff
@@ -92,22 +105,65 @@ func printVersion() {
 }
 
 var (
-	errNoPackageFound = errors.New("no package found")
+	errNoPackageFound        = errors.New("no package found")
+	errUnsupportMultiplePkgs = errors.New("multiple packages are not supported, please specify the package ID or directory path that only contains one package")
 )
 
 func generateFromRequest(request GenerationRequest) (string, error) {
 	cfg := packages.Config{
 		Mode: packages.NeedName | packages.NeedTypes,
 	}
-	pkgs, err := packages.Load(&cfg, request.Pkg)
+	if request.ModelPkg == "" || request.ModelPkg == request.Pkg {
+		pkgs, err := packages.Load(&cfg, request.Pkg)
+		if err != nil {
+			return "", err
+		}
+		if len(pkgs) < 1 {
+			return "", errNoPackageFound
+		}
+		if len(pkgs) > 1 {
+			return "", errUnsupportMultiplePkgs
+		}
+		return generator.GenerateRepositoryImpl(pkgs[0].Types, pkgs[0].Types, request.ModelName, request.RepoName, request.DestPkg)
+	}
+	intfPkgID, err := getPkgID(request.Pkg)
 	if err != nil {
 		return "", err
 	}
-	if len(pkgs) == 0 {
+	modelPkgID, err := getPkgID(request.ModelPkg)
+	if err != nil {
+		return "", err
+	}
+	patterns := []string{intfPkgID, modelPkgID}
+	pkgs, err := packages.Load(&cfg, patterns...)
+	if err != nil {
+		return "", err
+	}
+	if len(pkgs) != 2 {
```

Comment:
If case `modelPkg == repoPkg` is handled the same way as `modelPkg != repoPkg`, remove this check